service/src/MSRIOGroup.cpp

```diff
@@ -395,11 +399,114 @@ namespace geopm
                                                    IOGroup::M_UNITS_WATTS,
                                                    agg_function(msr_name),
                                                    ps.description + "\n    alias_for: " + ps.msr_name + " rate of change",
-                                                   IOGroup::M_SIGNAL_BEHAVIOR_VARIABLE};
+                                                   IOGroup::M_SIGNAL_BEHAVIOR_VARIABLE,
+                                                   string_format_double};
             }
         }
     }
 
+    void MSRIOGroup::register_pcnt_scalability_signals(void)
+    {
+        // Tracking for intermediate signals that will not
+        // be exposed to the user but will be used to 
+        // generate user visible signals
+        std::map<std::string, signal_info> signal_hidden;
+
+        // register time signal; domain board
+        std::string time_name = "MSR::TIME";
+        std::shared_ptr<Signal> time_sig = std::make_shared<TimeSignal>(m_time_zero, m_time_batch);
+        signal_hidden[time_name] = {std::vector<std::shared_ptr<Signal> >({time_sig}),
+                                    GEOPM_DOMAIN_CPU,
+                                    IOGroup::M_UNITS_SECONDS,
+                                    Agg::select_first,
+                                    "Time in seconds used to calculate pcnt_rate, acnt_rate",
+                                    IOGroup::M_SIGNAL_BEHAVIOR_VARIABLE,
+                                    string_format_double};
```

Comment:
Why is this all necessary?  I see that there's a ```MSR::TIME``` signal created above in register_power_signals().  Can that be used here?